### DISCUSSION TITLE
4Mb ESP8266

### DISCUSSION THREAD

**Discussion by bovineck** [forkineye/ESPixelStick#575] [Ideas]
I have a few ESP-01 modules around that I have upgraded to have 4Mb flash. I can upload V3.2 OK but is there an esptool command that would enable the 4-beta firmware to be uploaded? The Java uploader won't talk to it and doesn't have the ESP-01 option via ESPixelstick option anymore.

For anyone interested - the upgrade path is here: https://youtu.be/wmXeXsnMHxg

**Comment by MartinMueller2003** [comment]
I suspect you will need a custom platform to set the IO pins to those used in the ESPS V1/2 implementation. I would start with the ESP8266 generic platform profile and tweak it for use in this application.

The V4 flash tool should have no issues talking to this device. You will need to also create an entry in the flash tool HW config file to tell the flash tool what the platform can support.

**Reply by bovineck** [reply]
Thank you I’ll look into that 👍

**Comment by forkineye** [comment]
Are you putting the module into flash mode? This has to be done manually with ESP-01 modules unless you're using a board that takes care of it for you.

**Reply by bovineck** [reply]
Yes, I’m able to flash the modified esp8266 with v3.2 no problem. 👍😀

**Comment by MartinMueller2003** [comment]
Download this dist.zip file and give it a try. This version has a log capture/save to disk feature. Save the logs and attach them to this email chain.
    On Wednesday, November 2, 2022 at 03:36:18 PM EDT, bovineck ***@***.***> wrote:  
 
 


Yes, I’m able to flash the modified esp8266 with v3.2 no problem. 👍 😀

**Reply by bovineck** [reply]
Hi Martin, sorry how do I download the dist.zip file? Was it attached to the post?

**Comment by bovineck** [comment]
OK so first I erased the modified ESP-01 (programming mode):
`esptool.py --port /dev/ttyUSB1 erase_flash`

Then I unplugged and replugged (programming mode) and from the root directory of the firmware I used:
`esptool.py --chip esp8266 --baud 460800 --port /dev/ttyUSB1 --before default_reset --after hard_reset write_flash 0x0 firmware/esp8266/espsv3-app.bin 0x200000 firmware/filesystem.bin`

...which works.

After identifying the ESP-01 on the router via it's mac address I can login fine. I'll test with lights asap and report back.
